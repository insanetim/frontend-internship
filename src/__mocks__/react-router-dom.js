export const location = {}

export const useLocation = jest.fn(() => location)
export const useNavigate = jest.fn(() => jest.fn())
