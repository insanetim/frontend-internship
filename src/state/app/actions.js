import * as types from './types'

export const loadingOn = () => ({
  type: types.LOADING_ON
})

export const loadingOff = () => ({
  type: types.LOADING_OFF
})

export const showNotification = payload => ({
  type: types.SHOW_NOTIFICATION,
  payload
})

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})
