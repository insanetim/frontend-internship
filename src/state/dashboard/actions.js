import * as types from './types'

export const requestTrending = payload => ({
  type: types.REQUEST_TRENDING,
  payload
})

export const setTrending = payload => ({
  type: types.SET_TRENDING,
  payload
})

export const requestSearch = payload => ({
  type: types.REQUEST_SEARCH,
  payload
})

export const setSearch = payload => ({
  type: types.SET_SEARCH,
  payload
})

export const setSearchQuery = payload => ({
  type: types.SET_SEARCH_QUERY,
  payload
})

export const clearSearch = () => ({
  type: types.CLEAR_SEARCH
})
