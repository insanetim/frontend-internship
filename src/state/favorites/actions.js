import * as types from './types'

export const fetchFavorites = (payload = 1) => ({
  type: types.FETCH_FAVORITES,
  payload
})

export const setFavorites = payload => ({
  type: types.SET_FAVORITES,
  payload
})
