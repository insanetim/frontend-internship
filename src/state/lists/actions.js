import * as types from './types'

export const requestLists = (payload, cb) => ({
  type: types.REQUEST_LISTS,
  payload,
  cb
})

export const setLists = payload => ({
  type: types.SET_LISTS,
  payload
})

export const requestList = (payload, cb) => ({
  type: types.REQUEST_LIST,
  payload,
  cb
})

export const setList = payload => ({
  type: types.SET_LIST,
  payload
})

export const createList = (payload, cb) => ({
  type: types.CREATE_LIST,
  payload,
  cb
})

export const addToList = payload => ({
  type: types.ADD_TO_LIST,
  payload
})

export const removeFromList = payload => ({
  type: types.REMOVE_FROM_LIST,
  payload
})

export const deleteList = (payload, cb) => ({
  type: types.DELETE_LIST,
  payload,
  cb
})
