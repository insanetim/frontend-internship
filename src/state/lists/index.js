import listsReducer from './reducers'

export default listsReducer
