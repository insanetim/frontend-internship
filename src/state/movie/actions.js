import * as types from './types'

export const requestMovie = (payload, cb) => ({
  type: types.REQUEST_MOVIE,
  payload,
  cb
})

export const setMovie = payload => ({
  type: types.SET_MOVIE,
  payload
})

export const setMovieInFavorites = payload => ({
  type: types.SET_MOVIE_IN_FAVORITES,
  payload
})

export const setMovieInWatchlist = payload => ({
  type: types.SET_MOVIE_IN_WATCHLIST,
  payload
})

export const changeMovieInFavorites = payload => ({
  type: types.CHANGE_MOVIE_IN_FAVORITES,
  payload
})

export const changeMovieInWatchlist = payload => ({
  type: types.CHANGE_MOVIE_IN_WATCHLIST,
  payload
})
