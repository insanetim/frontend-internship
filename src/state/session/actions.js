import * as types from './types'

export const logIn = (payload, cb) => ({
  type: types.LOG_IN,
  payload,
  cb
})

export const logOut = cb => ({
  type: types.LOG_OUT,
  cb
})

export const setSession = payload => ({
  type: types.SET_SESSION,
  payload
})

export const deleteSession = () => ({
  type: types.DELETE_SESSION
})

export const fetchAccount = () => ({
  type: types.FETCH_ACCOUNT
})

export const setAccount = payload => ({
  type: types.SET_ACCOUNT,
  payload
})
