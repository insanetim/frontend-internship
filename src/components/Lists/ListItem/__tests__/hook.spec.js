import 'jsdom-global/register'
import { Modal } from 'antd'
import { useNavigate } from 'react-router-dom'
import { act, renderHook } from '@testing-library/react-hooks'

import { dispatch } from 'src/__mocks__/react-redux'
import { deleteList } from 'src/state/lists/actions'
import useContainer from '../hook'

describe('ListItem useContainer hook', () => {
  let result = null

  const navigate = jest.fn()
  useNavigate.mockReturnValue(navigate)

  jest.spyOn(Modal, 'confirm')

  beforeEach(() => {
    ;({ result } = renderHook(() => useContainer(1)))

    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot()
  })

  it('checks `handleClick` method', () => {
    act(() => {
      result.current.handleClick()
    })

    expect(navigate).toHaveBeenCalledWith('/list/1')
  })

  it('checks `handleDelete` method', () => {
    act(() => {
      result.current.handleDelete({ stopPropagation: jest.fn() })
    })

    expect(Modal.confirm).toHaveBeenCalled()
  })

  it('checks `onOk` method', () => {
    act(() => {
      result.current.onOk()
    })

    expect(dispatch).toHaveBeenCalledWith(deleteList(1))
  })
})
