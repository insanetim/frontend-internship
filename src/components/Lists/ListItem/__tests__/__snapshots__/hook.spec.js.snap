// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem useContainer hook matches snapshot 1`] = `
{
  "handleClick": [Function],
  "handleDelete": [Function],
  "onOk": [Function],
}
`;
